### PR TITLE
Enhanced config flag #2077

### LINKED ISSUE #2077: KICS - Configuration as Code
### Description
As a user, I'd like to be able to automate my KICS scans without the need for passing flags into the command line, but rather having it in the source code of my project in a specific file

**Use case scenario**
1. I create a JSON file named **kics.config** in the root of my project
2. In this file I write all the configuration needed for scanning my project with KICS (e.g., files to exclude, technology to scan)
3. KICS starts scan without flags (besides the relevant ones like the -p for instance)
4. KICS looks for kics.config file and consider its configurations
5. Scan runs as if we passed this config via cli flags

### PR DESCRIPTION
**Pull request by joaoReigota1**
Closes #2077

**Proposed Changes**

- Kics Can now be called without flags and will get the configurations from the 'kics.config' file inside the root of kics
-Kics will now check the contents of the config file and determine its extension 
-If there is no 'kics.config' file in the root of kics, kicks must be run with flags, or with the '--config' flag with the path to a configuration file 

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

**Comment by nunoocx** [comment]
There are 3 Low results in SAST scan. It seems that they are from your PR. Please Check https://sast.checkmarx.net/cxwebclient/ViewerMain.aspx?scanId=1052440&ProjectID=1199
image

File: internal/console/helpers/helpers.go [hunk 1]
```diff
@@ -176,3 +178,26 @@ func CustomConsoleWriter(fileLogger *zerolog.ConsoleWriter) zerolog.ConsoleWrite
...
+		return "yaml"
+	}
+
+	if c, err := hcl.Parse(string(rc)); err == nil {
+		if err = hcl.DecodeObject(&temp, c); err == nil {
+			return "hcl"
+		}
+	}
+
+	return "toml"
```
Comment (nunoocx) [inline internal/console/helpers/helpers.go:202]: What if I write something that is neither of the above and is not toml as well?
Maybe in that case we should report an error?

File: internal/console/helpers/helpers_test.go [hunk 2]
```diff
@@ -219,3 +220,64 @@ func TestProgressBar(t *testing.T) {
...
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := FileAnalyzer(tt.arg); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("FileAnalyzer() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
```
Comment (nunoocx) [inline internal/console/helpers/helpers_test.go:283]: Need to add a negative test: what if the content is neither of the supported formats? What is the behaviour?

**Review by nunoocx — CHANGES_REQUESTED** [review]
Check my comments ;)

**Review by nunoocx — APPROVED** [review]
LGTM

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**